Update lambda.py

ToROxI opened the pull request:
Update for module's logic to don't destroy pre-configured Lambda VPC settings if vpc_subnet_ids and vpc_security_group_ids not set. Explicit declaration of destruction instead.

##### SUMMARY
<!--- Describe the change, including rationale and design decisions -->

<!--- If you are fixing an existing issue, please include "Fixes #nnn" in your
commit message and your description; but you should still explain what
the change does.-->

##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->
 - Feature Pull Request
 - New Module Pull Request
 - Bugfix Pull Request
 - Docs Pull Request

##### COMPONENT NAME
<!--- Name of the module, plugin, module or task -->

##### ANSIBLE VERSION
<!--- Paste verbatim output from "ansible --version" between quotes below -->
```

```


##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here.
For bugs that don't have a linked bug report, a step-by-step reproduction
of the problem is helpful. -->

<!--- Paste verbatim command output below, e.g. before and after your change -->
```

```

ToROxI committed:
Update lambda.py

Update for module's logic to don't destroy pre-configured Lambda VPC settings if vpc_subnet_ids and vpc_security_group_ids not set. Explicit declaration of destruction instead.

ToROxI committed:
Update lambda.py

fix trailing whitespace

resmo commented:
I am -1 with the implementation, having a keywork 'None' is a no go. I even wonder, if the functionality is necessary as it can possible be achieved by using an empty list 


~~~yaml
# Explicit declaration of no-VPC configuration
- name: looped creation
  lambda:
    name: '{{ item.name }}'
    state: present
    zip_file: '{{ item.zip_file }}'
    runtime: 'python2.7'
    role: 'arn:aws:iam::987654321012:role/lambda_basic_execution'
    handler: 'hello_python.my_handler'
    vpc_subnet_ids: []
    vpc_security_group_ids: []
    environment_variables: '{{ item.env_vars }}'
    tags:
      key1: 'value1'
  with_items:
    - name: HelloWorld
      zip_file: hello-code.zip
      env_vars:
        key1: "first"
        key2: "second"
    - name: ByeBye
      zip_file: bye-code.zip
      env_vars:
        key1: "1"
        key2: "2"
~~~

needs_info

ToROxI commented:
@resmo, 
Hi! 
Main goal was to avoid destruction of VPC configuration if parameters are not provided in module.
So if keyword 'None' is not applicable for you - I'll remove this.

Please let me know if you have any concerns about my patch.
Thanks!

s-hertel reviewed (changes requested):
Right now both omitting the options from the playbook and setting the options to an empty list remove the current VPC configuration. Rather than setting a special 'None' keyword, it would be better to change that both omitting and setting to [] mean the same thing.
e.g.
* if vpc_security_group_ids and vpc_subnet_ids are both lists that contain things, set the VPC configuration
* if they are empty lists, remove the VPC configuration
* if they are None (module.params['vpc_subnet_ids'] == None when the option isn't provided in the task), leave the VPC configuration as it is

ToROxI committed:
Update lambda.py

Apply requested changes

ToROxI commented:
@s-hertel, requested change applied

ToROxI committed:
Update lambda.py

Update for logic of module not to force user to set runtime, handler and role if function already exists.

@s-hertel, please review this change.
Please let me know if you have any concerns about my patch.
Thanks!

ToROxI committed:
Update lambda.py

fix trailing whitespace

ToROxI committed:
Update lambda.py

fix logic statement

ToROxI committed:
Update lambda.py

fix logic issue

ToROxI committed:
Update lambda.py

fix logic issue 2

ToROxI committed:
Update lambda.py

change required_together

ToROxI commented:
@s-hertel , @achinthagunasekara 
there is need to update tests due to changes in the invocation sequence. Will you review test or I need to add this changes of test to PR?

Thanks!

ToROxI committed:
Update lambda.py

fix check for unsupported mutation operation

mattclay commented:
CI failure in integration tests: https://app.shippable.com/github/ansible/ansible/runs/73039/66/tests

gundalow commented:
Given feedback from resmo and shertel I'm going to close this.

For this to get in someone will need to raise a new PR that includes tests
